### PR TITLE
Houdini: Camera Loader fix mismatch for Maya cameras

### PR DESCRIPTION
**Pull request by MustafaJafar**
## Changelog Description
This PR adds 
- A workaround to match Maya render mask in Houdini
- `SetCameraResolution` inventory action
- set camera resolution when loading or updating camera

image

## Additional Info 1
Many thanks to @BigRoy, Code in this PR is mostly his..
https://github.com/BigRoy/OpenPype/blob/2810ee866066b55ad7f26ab2cebf2dc6de8978f2/openpype/hosts/houdini/plugins/load/load_camera.py

## Additional Info 2
This PR also fixes this issue [Set Camera resolution for Loaded camera in Houdini](https://community.ynput.io/t/set-camera-resolution-for-loaded-camera-in-houdini/412) (it was posted to forums)

## Testing notes:
1. Load alembic camera in houdini
2. Set version from `Manage` and check Aperture and Resolution
3. Test Inventory action `Set Camera Resolution`

### CODE REVIEW

File: openpype/hosts/houdini/plugins/load/load_camera.py [hunk 1]
```diff
@@ -90,13 +97,8 @@ class CameraLoader(load.LoaderPlugin):
...
     def load(self, context, name=None, namespace=None, data=None):
 
-        import os
-        import hou
-
         # Format file name, Houdini only wants forward slashes
-        file_path = self.filepath_from_context(context)
-        file_path = os.path.normpath(file_path)
-        file_path = file_path.replace("\\", "/")
+        file_path = self.fname.replace("\\", "/")
```
Comment (BigRoy) [inline openpype/hosts/houdini/plugins/load/load_camera.py:101]: Didn't we deprecate `self.fname` some time ago @iLLiCiTiT ?

Comment (MustafaJafar) [reply openpype/hosts/houdini/plugins/load/load_camera.py:101]: I found it's still defined

Comment (BigRoy) [reply openpype/hosts/houdini/plugins/load/load_camera.py:101]: Found it. It's this PR which tried to eradicate it as a first step: https://github.com/ynput/OpenPype/pull/4602

File: openpype/hosts/houdini/api/lib.py [hunk 2]
```diff
@@ -649,3 +649,60 @@ def get_color_management_preferences():
         "display": hou.Color.ocio_defaultDisplay(),
         "view": hou.Color.ocio_defaultView()
     }
+
+
+def get_current_asset_doc():
```
Comment (iLLiCiTiT) [inline openpype/hosts/houdini/api/lib.py:654]: Please do not duplicate already existing functions. There is `get_current_project_asset` in `openpype.pipeline.context_tools`.

File: openpype/hosts/houdini/plugins/inventory/set_camera_resolution.py [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+class SetCameraResolution(InventoryAction):
+
+    label = "Set Camera Resolution"
+    icon = "desktop"
+    color = "orange"
+
+    @staticmethod
+    def is_compatible(container):
+        print(container)
```
Comment (iLLiCiTiT) [inline openpype/hosts/houdini/plugins/inventory/set_camera_resolution.py:16]:
```suggestion
```

File: openpype/hosts/houdini/plugins/inventory/set_camera_resolution.py [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+        return (
+            container.get("loader") == "CameraLoader"
+        )
+
+    def process(self, containers):
+        for container in containers:
+            node = container["node"]
+            camera = get_camera_from_container(node)
+            set_camera_resolution(camera)
```
Comment (iLLiCiTiT) [inline openpype/hosts/houdini/plugins/inventory/set_camera_resolution.py:25]: Query the document from database only once.
```suggestion
        asset_doc = get_current_project_asset()
        for container in containers:
            node = container["node"]
            camera = get_camera_from_container(node)
            set_camera_resolution(camera, asset_doc)
```

**Review by antirotor — APPROVED** [review]
Works great! Not sure about the terminology though - asset document is really OpenPype term, not AYON one, but we can change it later, there are so many places where we'll need to do that anyway.

**Pull request merged**